content/Registration and Authentication in your Django app with dj-rest-auth part1- Bruno Michetti.md

```diff
@@ -0,0 +1,214 @@
+```
+
+In our settings we have to use the Django parameter called `AUTH_USER_MODEL` to specify our User model. If the `CustomUser` model is defined in an app called `users`, we must add:
+
+```python
+AUTH_USER_MODEL = 'users.CustomUser'
+```
+
+Besides, we have to add in our settings these parameters to use the email as the User identifier in our app:
+
+```python
+ACCOUNT_AUTHENTICATION_METHOD = 'email'
+ACCOUNT_USERNAME_REQUIRED = False
+ACCOUNT_EMAIL_REQUIRED = True
+ACCOUNT_UNIQUE_EMAIL = True
+```
+
+Also, let's configure by the moment a signup without email verification:
+
+```python
+ACCOUNT_EMAIL_VERIFICATION = 'none'
+```
+
+I'll talk about account verification in the second part of the article.  
+Now we need to add our custom serializer to define the two added attributes and use them in the registration. The custom register serializer needs to be a child class of the one named `RegisterSerializer` that dj-rest-auth provides. Also we have to overwrite the save method to set those values that came in the request:
+
+```python
+# serializers.py in the users Django app
+from django.db import transaction
+from rest_framework import serializers
+from dj_rest_auth.registration.serializers import RegisterSerializer
+
+from users.models import GENDER_SELECTION
+
+
+class CustomRegisterSerializer(RegisterSerializer):
+    gender = serializers.ChoiceField(choices=GENDER_SELECTION)
+    phone_number = serializers.CharField(max_length=30)
+
+    # Define transaction.atomic to rollback the save operation in case of error
+    @transaction.atomic
+    def save(self, request):
+        user = super().save(request)
+        user.gender = self.data.get('gender')
+        user.phone_number = self.data.get('phone_number')
+        user.save()
+        return user
+
+```
+
+Now we can't forget to tell dj-rest-auth to use the recently created serializer. Let's add the `CustomRegisterSerializer` to the settings:
+
+```python
+REST_AUTH_REGISTER_SERIALIZERS = {
+    'REGISTER_SERIALIZER': 'users.serializers.CustomRegisterSerializer',
+}
+```
+
+After this, if we go to the registration endpoint, we would see something like this:
+
+<img src="images/dj-rest-auth-registration-example.png" alt="Registration endpoint" width="900"/>
+
+The username appears in the browsable endpoint, but thanks to the configuration you can leave it empty.  
+And thats it! we have added a customized registration endpoint to our Django app. Now we can create users with email, gender and phone number, and test the login too.
+
+### Use of JWT
+
+When an app manages authentication, a good and standard way is using [JWT (JSON web tokens)](https://auth0.com/docs/tokens/json-web-tokens). JWTs are a good way of securely transmitting information between parties because they can be signed, which means you can be sure that the senders are who they say they are. We can easily configure our Django app to use JWT. First we need to install `djangorestframework-simplejwt`:
```

Comment:
```suggestion
When an app manages authentication, a good and standard implementation of authentication by token mechanism is using [JWT (JSON web tokens)](https://auth0.com/docs/tokens/json-web-tokens). JWTs are a good way of securely transmitting information between parties because they can be signed, which means you can be sure that the senders are who they say they are. We can easily configure our Django app to use JWT. First we need to install `djangorestframework-simplejwt`:
```

Reply:
Awesome